src/mlpack/methods/reinforcement_learning/training_config.hpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file training_config.hpp
+ * @author Shangtong Zhang
+ *
+ * This file is the implementation of TrainingConfig class,
+ * which contains hyper-parameters for training RL agent.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_TRAINING_CONFIG_HPP
+#define MLPACK_METHODS_RL_TRAINING_CONFIG_HPP
+
+namespace mlpack {
+namespace rl {
+
+class TrainingConfig
```

Comment:
Sorry for the slow response here---I was unavailable over the weekend so I could only get to this now.  I think for the hyperparameter tuner, we would only need to add a constructor to `TrainingConfig` that also takes all of the parameter values.  I believe the HP tuner will be able to pass hyperparameters to "nested" constructors, so even if not all hyperparameters are directly set in the top-level RL class constructors, it's ok if they are set in constructors of objects that are passed to RL classes.